bundles/statistics/statsgrid2016/components/Legend.js

```diff
@@ -69,17 +69,24 @@ Oskari.clazz.define('Oskari.statistics.statsgrid.Legend', function(sandbox, loca
             // attach container to parent if provided, otherwise updates UI in the current parent
             el.append(container);
         }
+
+        var renderDone = function(){
+            me._renderDone();
+            if(typeof handler === 'function') {
```

Comment:
I'd say you should send the optional handler to me._renderDone(handler) OR attach it to renderState. Having 2 similar function names isn't too much fun when trying to figure out what's going on. Should handler be called even with immediate repaint? If you attach the handler to renderState you can call it after rendering (including repaint) is totally finished. The again having another call to render with handler would overwrite the current... So maybe the best option would be to make the Legend component observable and trigger 'change' etc events after render is completed:

    _renderDone : function() {
        var state = this._renderState;
        this._renderState = {};
        this._restorePanelState(this._accordion, state.panels);
        if(state.repaint) {
            this.render(state.el);
        } else {
            this.trigger('change');
       }
    },